packages/ibm-products/src/components/OptionsTile/OptionsTile.tsx

```diff
@@ -26,60 +21,132 @@ import {
   WarningFilled,
 } from '@carbon/react/icons';
 import * as carbonMotion from '@carbon/motion';
+import { CarbonIconType } from '@carbon/icons-react/lib/CarbonIcon';
 
-// The block part of our conventional BEM class names (blockClass__E--M).
 const blockClass = `${pkg.prefix}--options-tile`;
 const componentName = 'OptionsTile';
 
-// NOTE: the component SCSS is not imported here: it is rolled up separately.
+interface OptionsTileProps {
+  /**
+   * Provide content to render as expandable OptionsTile. If no children
+   * are present, the OptionsTile will render as its variant.
+   */
+  children?: ReactNode;
 
-// Default values for props
-const defaults = {
-  onChange: () => {},
-  size: 'xl',
-};
+  /**
+   * Provide an optional class to be applied to the containing node.
+   */
+  className?: string;
+
+  /**
+   * Whether the toggle is enabled or disabled. If nothing is passed,
+   * no toggle will be rendered.
+   */
+  enabled?: boolean;
+
+  /**
+   * Whether the OptionsTile is in invalid validation state.
+   */
+  invalid?: boolean;
+
+  /**
+   * Provide a text explaining why the OptionsTile is in invalid state.
+   */
+  invalidText?: string;
+
+  /**
+   * Whether the OptionsTile is in locked validation state.
+   */
+  locked?: boolean;
+
+  /**
+   * Provide a text explaining why the OptionsTile is in locked state.
+   */
+  lockedText?: string;
+
+  /**
+   * Provide a function which will be called each time the user
+   * toggles the open state of the OptionsTile.
+   */
+  onChange?: (value: any) => void;
+
+  /**
+   * Provide a function which will be called each time the user
+   * interacts with the toggle.
+   */
+  onToggle: (value: any) => void;
```

Comment:
Same suggested as above as the onToggle prop returns a boolean as well
```suggestion
  onToggle: (value: boolean) => void;
```